app/middlewares/session_validation.js

```diff
@@ -0,0 +1,28 @@
+const errors = require('../errors');
+const jwt = require('jwt-simple');
+const {
+  common: {
+    session: { secret }
+  }
+} = require('../../config');
+const { findUser } = require('../services/users');
+
+exports.checkSession = async (req, _, next) => {
+  const token = req.headers.authorization;
+  try {
+    if (token) {
+      const { email } = jwt.decode(token, secret);
+      if (email) {
+        const user = await findUser({ email });
+        if (user) {
+          return next();
+        }
+        throw errors.unauthorizedError('Invalid Token');
+      }
+      throw errors.unauthorizedError('Invalid Token');
+    }
+    throw errors.unauthorizedError('Invalid Token');
```

Comment:
you are throwing an error instead of returning next(error)
this could be fine if you return these errors in the catch, but there you are putting again the same error instead of using the one passed by parameter
I would use return next(error), is the cleanest way

Reply:
suppose the `findUser` throw a database error, you are sending an unathorizedError which is not true